pkg/tenant/api/v1/client.go

```diff
@@ -186,119 +186,89 @@ func (s *APIv1) TenantDelete(ctx context.Context, id string) (err error) {
 	return nil
 }
 
-func (s *APIv1) AppList(ctx context.Context, in *AppQuery) (out *AppPage, err error) {
-	// Make the HTTP request
-	var req *http.Request
-	if req, err = s.NewRequest(ctx, http.MethodGet, "/v1/apps", nil, nil); err != nil {
+func (s *APIv1) TenantMemberList(ctx context.Context, id string, in *PageQuery) (out *TenantMemberPage, err error) {
+	if id == "" {
 		return nil, err
 	}
 
-	out = &AppPage{}
-	if _, err = s.Do(req, out, true); err != nil {
-		return nil, err
-	}
+	path := fmt.Sprintf("v1/tenant/%s/members", id)
 
-	return out, nil
-}
+	var params url.Values
+	if params, err = query.Values(in); err != nil {
+		return nil, fmt.Errorf("could not encode query params: %w", err)
+	}
 
-func (s *APIv1) AppCreate(ctx context.Context, in *App) (out *App, err error) {
-	// Make the HTTP request
 	var req *http.Request
-	if req, err = s.NewRequest(ctx, http.MethodPost, "/v1/apps", nil, nil); err != nil {
+	if req, err = s.NewRequest(ctx, http.MethodGet, path, nil, &params); err != nil {
 		return nil, err
 	}
 
-	out = &App{}
+	out = &TenantMemberPage{}
 	if _, err = s.Do(req, out, true); err != nil {
 		return nil, err
 	}
 
 	return out, nil
 }
 
-func (s *APIv1) AppDetail(ctx context.Context, id string) (out *App, err error) {
-	// Make the HTTP request
-	var req *http.Request
-	if req, err = s.NewRequest(ctx, http.MethodGet, "/v1/apps/:id", id, nil); err != nil {
-		return nil, err
-	}
-
-	out = &App{}
-	if _, err = s.Do(req, out, true); err != nil {
-		return nil, err
+func (s *APIv1) TenantMemberCreate(ctx context.Context, id string, in *TenantMember) (err error) {
+	if id == "" {
+		return err
 	}
 
-	return out, nil
-}
+	path := fmt.Sprintf("v1/tenant/%s/members", id)
 
-func (s *APIv1) AppDelete(ctx context.Context, id string) (err error) {
-	// Make the HTTP request
 	var req *http.Request
-	if req, err = s.NewRequest(ctx, http.MethodDelete, "/v1/apps/:id", id, nil); err != nil {
+	if req, err = s.NewRequest(ctx, http.MethodPost, path, in, nil); err != nil {
 		return err
 	}
 
-	if _, err = s.Do(req, nil, true); err != nil {
+	var rep *http.Response
+	if rep, err = s.Do(req, nil, true); err != nil {
 		return err
 	}
-	return nil
-}
-
-func (s *APIv1) TopicList(ctx context.Context, in *TopicQuery) (out *TopicPage, err error) {
-	// Make the HTTP request
-	var req *http.Request
-	if req, err = s.NewRequest(ctx, http.MethodGet, "/v1/topics", in, nil); err != nil {
-		return nil, err
-	}
 
-	out = &TopicPage{}
-	if _, err = s.Do(req, out, true); err != nil {
-		return nil, err
+	if rep.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("expected no content, received %s", rep.Status)
```

Comment:
Good check! If we do have this endpoint return a `Member` object, then you can check for `StatusCreated` (201) instead.